Update for Layout/FirstParameterIndentation name change

Floppy opened the pull request:
Resolves #42 

Rubocop 0.68 renames `Layout/FirstParameterIndentation` to `Layout/IndentFirstArgument`. Frustratingly, 0.67 doesn't support the newer name, and 0.68 doesn't support the older one, so this is a breaking change on their part with no migration path. Because of that, I've had to update the minimum rubocop version to 0.68 in this PR as well. I appreciate you might not want to do that though...

Floppy commented:
Apparently the cop was split in two, not just renamed, so I need to add the other half. I'll have to do that later on this evening though.

koic commented on config/rails.yml:
```diff
@@ -77,9 +77,12 @@ Layout/EmptyLinesAroundMethodBody:
+  
+Layout/IndentFirstParameter:
+  Enabled: true
+  
```
It seems that there is an unnecessary space before and after.

```diff
--- a/config/rails.yml
+++ b/config/rails.yml
@@ -79,10 +79,10 @@ Layout/EmptyLinesAroundModuleBody:

 Layout/IndentFirstArgument:
   Enabled: true
-
+
 Layout/IndentFirstParameter:
   Enabled: true
-
+
 # Use Ruby >= 1.9 syntax for hashes. Prefer { a: :b } over { :a => :b }.
 Style/HashSyntax:
   Enabled: true
```

koic commented on Appraisals:
```diff
@@ -1,34 +1,6 @@
-appraise "rubocop-0.67" do
-  gem "rubocop", "~> 0.67"
+appraise "rubocop-0.68" do
+  gem "rubocop", "~> 0.68"
```
I think that it is good to add RuboCop Performance.

```ruby
gem "rubocop-performance"
```

toshimaru replied:
@koic Do you think we should also add `rubocop-performance` as the gem's runtime dependency?

koic replied:
@toshimaru Sure! It is reasonable to add to gemspec.

Floppy replied:
Done!

koic commented on Appraisals:
```diff
@@ -1,34 +1,6 @@
-  gem "rails"
-end
-appraise "rubocop-0.67" do
-  gem "rubocop", "~> 0.67"
```
If these are dropped, then the files from `gemfiles/rubocop_0.60.gemfile` to `gemfiles/rubocop_0.67.gemfile` will also need to be removed.

Floppy replied:
Done!

koic commented:
FYI, I wrote some comments. On the other hand, rails/rails is waiting for the upgrade of Code Climate.
https://github.com/rails/rails/pull/36176

Also, wait for the owner's decision as to whether RuboCop 0.67 or lower will be dropped.
Cc @toshimaru

toshimaru commented:
In my plan, after updating to rubocop v0.68 in Rails official repository, It'd be good time to drop the support of rubocop v0.67- and Ruby 2.3.

So, current status is **waiting for rubocop update in rails/rails**.

Floppy commented:
Sounds sensible @toshimaru - I'll update with @koic's suggested changes and then when Rails upgrades, this PR will be ready.

stas commented:
It would be great to move forward with this if possible. Thanks in advance :bowing_man:

j-nguyen commented:
Any updates? @koic @toshimaru

toshimaru commented:
@Floppy could you fix @koic 's poiint? 

- https://github.com/toshimaru/rubocop-rails_config/pull/43#discussion_r281504612
- https://github.com/toshimaru/rubocop-rails_config/pull/43#discussion_r281504525
- https://github.com/toshimaru/rubocop-rails_config/pull/43#discussion_r281504590

toshimaru commented:
I've noticed Rails rubocop version is still using 0.67, so I can't get this merged yet... 😞 

https://github.com/rails/rails/blob/f0445213d8f23c982d1c080a750b33d4855d46e4/.codeclimate.yml#L25-L28

koic commented:
https://github.com/rails/rails/pull/36426 has been merged. Could you update this PR?

koic commented on config/rails.yml:
```diff
@@ -77,9 +77,12 @@ Layout/EmptyLinesAroundMethodBody:
   Enabled: true
 
-Layout/FirstParameterIndentation:
+Layout/IndentFirstArgument:
```
I think it is preferable to move in ASCII order.

Floppy replied:
Done!

koic commented on config/rails.yml:
```diff
@@ -77,9 +77,12 @@ Layout/EmptyLinesAroundMethodBody:
-
+  
+Layout/IndentFirstParameter:
+  Enabled: true
```
This cop is not added in rails/rails.
https://github.com/rails/rails/pull/36426

Floppy replied:
Gone!

koic commented:
And rails/rails has been updated to RuboCop 0.71.

Floppy commented:
Ah! Great! I'll get this finished off.

Floppy commented:
OK, I've updated this to completely match the new Rails config, including adding the dependency on `rubocop-rails`. I've also added appraisals up to rubocop 0.71.

Floppy commented:
Interesting, this is now throwing a style error in bin/yarn in new rails apps. Hmm.

Floppy commented:
OK, this is a style error in the rails template at https://github.com/rails/rails/blob/b2eb1d1c55a59fee1e6c4cba7030d8ceb524267c/railties/lib/rails/generators/rails/app/templates/bin/yarn.tt.

How should I deal with that?

koic commented on .travis.yml:
```diff
@@ -10,11 +10,7 @@ rvm:
+  - gemfiles/rubocop_0.68.gemfile
+  - gemfiles/rubocop_0.69.gemfile
+  - gemfiles/rubocop_0.70.gemfile
+  - gemfiles/rubocop_0.71.gemfile
```
EOF is forgotten.

Floppy replied:
Fixed

koic commented on config/rails.yml:
```diff
@@ -1,5 +1,9 @@
+
 AllCops:
-  TargetRubyVersion: 2.2
+  TargetRubyVersion: 2.5
```
rubocop-rails_config supports Ruby 2.3 or higher.
https://github.com/toshimaru/rubocop-rails_config/blob/v0.5.1/rubocop-rails_config.gemspec#L13

I think it will be 2.3 if `TargetRubyVersion` will be updated.

Cc @toshimaru

Floppy replied:
OK - I've changed it to 2.3. That was just a straight copy from the Rails config though, so the version will be different.

Floppy commented:
Oh, that's fixed the `bin/yarn` problem in the tests too... that's unexpected.

toshimaru commented on rubocop-rails_config.gemspec:
```diff
@@ -12,8 +12,10 @@ Gem::Specification.new do |spec|
-  spec.add_dependency "rubocop", "~> 0.60"
+  spec.add_dependency "rubocop", "~> 0.68"
   spec.add_dependency "railties", ">= 3.0"
+  spec.add_dependency "rubocop-performance", "~> 1.3"
```
:+1:

koic commented on rubocop-rails_config.gemspec:
```diff
@@ -12,8 +12,10 @@ Gem::Specification.new do |spec|
   spec.required_ruby_version = ">= 2.3.0"
 
-  spec.add_dependency "rubocop", "~> 0.60"
+  spec.add_dependency "rubocop", "~> 0.68"
```
RuboCop Rails 2.0 requires RuboCop >= 0.70.
https://rubygems.org/gems/rubocop-rails/versions/2.0.1

RuboCop 0.68 and 0.69 will not be required. So I think it is better to require RuboCop 0.70 or higher.

```diff
- spec.add_dependency "rubocop", "~> 0.68"
+ spec.add_dependency "rubocop", "~> 0.70"
```

RuboCop 0.68 and 0.69 can also be dropped from .travis.yml and Appraisal.

Cc @toshimaru

Floppy replied:
Done!

koic commented on rubocop-rails_config.gemspec:
```diff
@@ -12,8 +12,10 @@ Gem::Specification.new do |spec|
+  spec.add_dependency "rubocop", "~> 0.70"
   spec.add_dependency "railties", ">= 3.0"
+  spec.add_dependency "rubocop-performance", "~> 1.3"
+  spec.add_dependency "rubocop-rails", "~> 2.0"
```
This is a small point. It may be helpful to summarize the dependency on RuboCop.

```diff
- spec.add_dependency "rubocop", "~> 0.60"
+ spec.add_dependency "rubocop", "~> 0.70"
+ spec.add_dependency "rubocop-performance", "~> 1.3"
+ spec.add_dependency "rubocop-rails", "~> 2.0"
  spec.add_dependency "railties", ">= 3.0"
```

Floppy replied:
yes, that's neater. Done.

koic commented:
@Floppy Thank you for updating!
@toshimaru I merged this PR. Can you release the new rubocop-rails_config?

toshimaru commented:
[v0.6.0](https://github.com/toshimaru/rubocop-rails_config/releases/tag/v0.6.0) has been release.

Thanks you so much for your effort. @koic @Floppy
